src/ORM/DataObject.php

```diff
@@ -1582,6 +1741,52 @@ public function getComponent($componentName)
         return $component;
     }
 
+    /**
+     * Assign an item to the given component
+     *
+     * @param string $componentName
+     * @param DataObject|null $item
+     * @return $this
+     */
+    public function setComponent($componentName, $item)
```

Comment:
Given the `getComponent/setComponent` symmetry, would it make sense to introduce `setComponents` in `getComponents/setComponents`? Probably not, just wondering if you had considered it.

Reply:
It's a bit harder, because we don't assign relations directly, we'd need to convert values from SS_List -> internal relation. It's out of scope of this change.

Reply:
I think we could implement a `Relation::copyTo(Relation $dest)` in 5.0

Reply:
All good, just wanted to hear your take on it